vllm/executor/distributed_gpu_executor.py

```diff
@@ -64,16 +64,18 @@ def initialize_cache(self, num_gpu_blocks: int,
                           num_cpu_blocks=num_cpu_blocks)
 
     def execute_model(
-        self, execute_model_req: ExecuteModelRequest
-    ) -> Optional[List[SamplerOutput]]:
+            self,
+            execute_model_req: ExecuteModelRequest) -> List[SamplerOutput]:
         if self.parallel_worker_tasks is None:
             self.parallel_worker_tasks = self._run_workers(
                 "start_worker_execution_loop",
                 async_run_tensor_parallel_workers_only=True,
                 **self.extra_execute_model_run_workers_kwargs)
 
         # Only the driver worker returns the sampling results.
-        return self._driver_execute_model(execute_model_req)
+        driver_outputs = self._driver_execute_model(execute_model_req)
+        assert driver_outputs is not None
```

Comment:
what's this change for? If we should assert it, I think we should also change the output of `_driver_execute_model ` to be non optional and we probably don't need assert (the mypy should catch this)?

Reply:
I think although the return value is not `None` for the default implementation (`DistributedGPUExecutor`), the child classes (e.g., `MultiprocessingGPUExecutor`) may have implementations that still return `None`?